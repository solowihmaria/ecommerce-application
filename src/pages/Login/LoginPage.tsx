import styles from './LoginPage.module.scss';
import { LoginForm } from '../../components/blocks/LoginForm/LoginForm';

export const LoginPage = () => {
    return (
        <div data-testid="test-id-login-page" className={styles.loginPage}>
            <LoginForm />
        </div>
    );
};
