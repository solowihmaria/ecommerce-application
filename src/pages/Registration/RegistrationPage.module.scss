@use '../../assets//styles/constants.module' as c;

.registrationPage {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    background: c.$color-accent-dark;
    background: url('../../assets/images/form-bg.jpg') no-repeat center center;
    background-size: cover;
}

.registrationContent {
    display: flex;
    flex: 1;
    align-items: center;
    justify-content: center;
    padding: 4rem 2rem;
}
