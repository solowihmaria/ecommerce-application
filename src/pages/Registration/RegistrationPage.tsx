import styles from './RegistrationPage.module.scss';
import { RegistrationForm } from '../../components/blocks/RegistrationForm/RegistrationForm';
import { Header } from '../../components/blocks/Header/Header';

export const RegistrationPage = () => {
    return (
        <div
            data-testid="test-id-registration-page"
            className={styles.registrationPage}
        >
            <Header />
            <main className={styles.registrationContent}>
                <RegistrationForm />
            </main>
        </div>
    );
};
