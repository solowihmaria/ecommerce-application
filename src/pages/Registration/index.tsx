import React from 'react';
const RegistrationPage: React.FC = () => {
    return (
        <div>
            <h1>Страница регистрации</h1>
        </div>
    );
};

export default RegistrationPage;
