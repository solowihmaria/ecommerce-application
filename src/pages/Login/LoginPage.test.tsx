import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LoginPage } from './LoginPage';
import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { RegistrationPage } from '../Registration/RegistrationPage';

describe('<LoginPage/> component', () => {
    test('Render LoginPage component', () => {
        render(
            <MemoryRouter>
                <LoginPage />
            </MemoryRouter>
        );
        const loginPage = screen.getByTestId('test-id-login-page');
        expect(loginPage).toBeInTheDocument();
        expect(loginPage).toMatchSnapshot();
    });
    test('Render LoginPage component - click Sign Up Link', async () => {
        render(
            <MemoryRouter initialEntries={['/login']}>
                <Routes>
                    <Route path="/login" element={<LoginPage />} />
                    <Route path="/register" element={<RegistrationPage />} />
                </Routes>
            </MemoryRouter>
        );
        const link = screen.getByTestId('sign-up-link');
        await userEvent.click(link);
        const registrationPage = screen.getByTestId(
            'test-id-registration-page'
        );
        expect(registrationPage).toBeInTheDocument();
    });
});
