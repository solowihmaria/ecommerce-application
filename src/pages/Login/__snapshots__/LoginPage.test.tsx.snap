// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage/> component Render LoginPage component 1`] = `
<div
  class="loginPage"
  data-testid="test-id-login-page"
>
  <header
    class="headerWrapper"
    data-testid="test-id-header"
  >
    <nav
      class="header"
    >
      <div
        class="leftSection"
      >
        <a
          class="logoLink"
          data-discover="true"
          href="/main"
        >
          <svg
            data-testid="mock-svg"
          />
        </a>
      </div>
      <div
        class="centerSection"
      >
        <a
          data-discover="true"
          href="/main"
        >
          <button
            class="button ghost navButton"
          >
            Main
          </button>
        </a>
        <a
          data-discover="true"
          href="/catalog"
        >
          <button
            class="button ghost navButton"
          >
            Catalog
          </button>
        </a>
        <a
          data-discover="true"
          href="/"
        >
          <button
            class="button ghost navButton"
          >
            About
          </button>
        </a>
        <a
          data-discover="true"
          href="/login"
        >
          <button
            class="button ghost authButton"
          >
            Sign In
          </button>
        </a>
        <a
          data-discover="true"
          href="/register"
        >
          <button
            class="button ghost authButton"
          >
            Sign Up
          </button>
        </a>
      </div>
      <div
        class="rightSection"
      >
        <a
          class="cartLink"
          data-discover="true"
          href="/"
        >
          <svg
            data-testid="mock-svg"
          />
        </a>
      </div>
    </nav>
  </header>
  <main
    class="loginContent"
  >
    <div
      class="loginContainer"
    >
      <form
        class="form"
        data-testid="test-id-login-form"
      >
        <h2
          class="heading heading--h2 formTitle"
        >
          Login
        </h2>
        <div
          class="formGroup"
        >
          <label
            class="label required"
            for="email"
          >
            Email
            <span
              aria-hidden="true"
            >
              *
            </span>
          </label>
          <div
            class="inputContainer"
          >
            <div
              class="inputWrapper"
            >
              <input
                autocomplete="email"
                class="input"
                data-testid="test-id-input-field"
                id="email"
                name="email"
                placeholder="Enter your email"
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          class="formGroup"
        >
          <label
            class="label required"
            for="password"
          >
            Password
            <span
              aria-hidden="true"
            >
              *
            </span>
          </label>
          <div
            class="inputContainer"
          >
            <div
              class="inputWrapper"
            >
              <input
                class="input password"
                id="password"
                name="password"
                placeholder="Enter your password"
                type="password"
              />
              <button
                aria-label="Show password"
                class="toggleButton"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <button
          class="button primary submitButton"
          type="submit"
        >
          Log In
        </button>
        <div
          class="signUpRedirect"
        >
          <p
            class="signUpText"
          >
            Don't have an account?
          </p>
          <a
            class="signUpLink"
            data-discover="true"
            data-testid="sign-up-link"
            href="/register"
          >
            Sign Up
          </a>
        </div>
      </form>
    </div>
  </main>
</div>
`;
